### PR TITLE
CAMEL-12004 Add isDebugEnabled() guard for debug level logs

### PR DESCRIPTION
**Pull request by mehranhassani**
I added the debug level guards to the one that I found with my script in the latest revision of source code.

### CODE REVIEW

**Comment by PascalSchumacher** [comment]
Thanks for the pull request.

Imho it would be better to convert the statements to parametrized log messages (where possible). That also saves the cost of the string concatenation (I do not see any potentially costly parameter creation) and does not add the complexity of an additional if statement to the code.

**Comment by mehranhassani** [comment]
I think the best way to deal with this problem is to use slf4j. But I don't know how costly the migration will be. I can look into the changes and try to parameterize them if its possible if this is the best solution in your opinion.

File: components/camel-spring/src/main/java/org/apache/camel/spring/spi/TransactionErrorHandler.java [hunk 1]
```diff
@@ -154,7 +154,7 @@ protected void processInTransaction(final Exchange exchange) throws Exception {
             if (log.isDebugEnabled()) {
                 // log exception if there was a cause exception so we have the stack trace
                 Exception cause = exchange.getException();
-                if (cause != null) {
+                if (cause != null && log.isDebugEnabled()) {
```
Comment (davsclaus) [inline components/camel-spring/src/main/java/org/apache/camel/spring/spi/TransactionErrorHandler.java]: log debug enable is already checked above

File: camel-core/src/test/java/org/apache/camel/LanguageTestSupport.java [hunk 2]
```diff
@@ -94,8 +94,9 @@ protected Object evaluateExpression(String expressionText, String expectedValue)
         } else {
             value = expression.evaluate(exchange, Object.class);
         }
-
-        log.debug("Evaluated expression: {} on exchange: {} result: {}", expression, exchange, value);
+        if (log.isDebugEnabled()) {
```
Comment (davsclaus) [inline camel-core/src/test/java/org/apache/camel/LanguageTestSupport.java]: This is not needed when using {}

File: components/camel-jms/src/main/java/org/apache/camel/component/jms/reply/QueueReplyManager.java [hunk 3]
```diff
@@ -131,7 +131,9 @@ protected AbstractMessageListenerContainer createListenerContainer() throws Exce
                 answer = new SharedQueueMessageListenerContainer(endpoint, fixedMessageSelector);
                 // must use cache level consumer for fixed message selector
                 answer.setCacheLevel(DefaultMessageListenerContainer.CACHE_CONSUMER);
-                log.debug("Using shared queue: " + endpoint.getReplyTo() + " with fixed message selector [" + fixedMessageSelector + "] as reply listener: " + answer);
+                if (log.isDebugEnabled()) {
```
Comment (davsclaus) [inline components/camel-jms/src/main/java/org/apache/camel/component/jms/reply/QueueReplyManager.java]: We could use `{}` style here

File: components/camel-test/src/main/java/org/apache/camel/test/junit4/LanguageTestSupport.java [hunk 4]
```diff
@@ -85,8 +85,9 @@ protected void assertExpression(String expressionText, String expectedValue, Str
         } else {
             value = expression.evaluate(exchange, Object.class);
         }
-
-        log.debug("Evaluated expression: " + expression + " on exchange: " + exchange + " result: " + value);
+        if (log.isDebugEnabled()) {
```
Comment (davsclaus) [inline components/camel-test/src/main/java/org/apache/camel/test/junit4/LanguageTestSupport.java]: The rest is just unit tests and tooling so its not as important as runtime camel-core etc. However the changes are fine to do, but maybe favor using `{}` style

Comment (mehranhassani) [reply components/camel-test/src/main/java/org/apache/camel/test/junit4/LanguageTestSupport.java]: In general, you do not recommend increasing complexity with the if statement. right?

**Pull request closed without merging**